packages/agoric-cli/src/commands/wallet.js

```diff
@@ -0,0 +1,103 @@
+// @ts-check
+/* eslint-disable func-names */
+/* global fetch, process */
+import { normalizeBech32 } from '@cosmjs/encoding';
+import { execSync } from 'child_process';
+import { Command } from 'commander';
+import { fmtRecordOfLines, simplePurseBalances } from '../lib/format.js';
+import { simpleOffers } from '../lib/psm.js';
+import { makeRpcUtils, networkId, networks } from '../lib/rpc.js';
+import { getWalletState } from '../lib/wallet.js';
+
+const normalizeAddress = literalOrName => {
+  try {
+    return normalizeBech32(literalOrName);
+  } catch (_) {
+    // not an address so assume it's a key
+    const buff = execSync(`agd keys show --address ${literalOrName}`);
+    return normalizeBech32(buff.toString().trim());
+  }
+};
+
+export const makeWalletCommand = async () => {
+  const wallet = new Command('wallet').description('wallet commands');
+
+  wallet
+    .command('send')
+    .description('send a prepared offer')
+    .requiredOption(
+      '--from [address]',
+      'address literal or name',
+      normalizeAddress,
+    )
+    .requiredOption('--offer [filename]', 'path to file with prepared offer')
+    .action(function () {
+      const { from, offer } = this.opts();
+      const { chainId, rpc } = networks[networkId];
+
+      const cmd = `agd --node=${rpc} --chain-id=${chainId} --from=${from} tx swingset wallet-action --allow-spend "$(cat ${offer})"`;
```

Comment:
can we pipe in --home and --keyring-backend as well from the root commnand?

Reply:
you got it: [pass through agd opts in perf command](https://github.com/Agoric/agoric-sdk/pull/6176/commits/03288963e5746186d8132a156e1ca44db85177d7)